### ISSUE TITLE
Sessions not loading from database when server restarts?

### ISSUE THREAD

**Issue by brunchboy** [luminus-framework/jdbc-ring-session#19]
Greetings! I configured the project I am starting to use `jdbc-ring-session` for its session store because I want my sessions to survive restarts of the server (otherwise I would just use the default in-memory sessions). And I can see that the database is being updated with sessions correctly. But whenever I restart the server, the first request that comes in using a particular `ring-session` header causes that session row in the database to be updated to hold a `nil` session, meaning my browsers all get logged out whenever the server restarts, the opposite of what I want to happen. They continue to use the same `ring-session` header value, and when I log in the database row gets updated to once again hold proper session state, until the next time the server restarts, after which time it gets wiped out by the next request again.

I must be misunderstanding something fundamental. How do I get my sessions to persist past restarts of the web application?

In case it is of any help, my source code is here: https://github.com/brunchboy/shade

**Comment by yogthos** [comment]
Session information will persist server restarts if the sessions are configured properly. One of the problems I see is that you're wrapping multiple session middleware:

```clojure
(defn wrap-base [handler]
  (-> ((:middleware defaults) handler)
      wrap-auth
      wrap-flash
       ;; this is a session from undertow
      (wrap-session {:timeout      Integer/MAX_VALUE
                     :cookie-attrs {:max-age   Integer/MAX_VALUE
                                    :http-only false
                                    :same-site true}})
      ;; and here the ring session is wrapped
      (wrap-defaults (-> site-defaults
                         (assoc-in [:security :anti-forgery] false)  ; TODO: Why can't I enable this?
                         (assoc-in [:session :store] (jdbc-store db/*db*))
                         (assoc-in [:session :cookie-attrs] {:max-age   Integer/MAX_VALUE
                                                             :http-only false
                                                             :same-site :strict})))
      wrap-internal-error))
```

The session store simply checks for the key generated by ring session [here](https://github.com/luminus-framework/jdbc-ring-session/blob/master/src/jdbc_ring_session/core.clj#L70), so as long as the key from the browser cookie matches then the session will be loaded.

**Comment by brunchboy** [comment]
Ah, thank you! I thought something like might be happening, especially since I saw two cookies being set, but the ring session configured by Luminus was not clear to me. I will try taking out the undertow `wrap-session` and see if that sorts things out. Is that the direction you’d suggest? I’ll likely be able to close this soon. Thank you so much for responding, even though this is clearly not an issue with your code. I tried asking for help on both the Luminus and ring channels in Clojurians to no avail.
